src/app/redux/AppReducer.test.js

```diff
@@ -0,0 +1,173 @@
+import chai, { expect } from 'chai';
+import chaiImmutable from 'chai-immutable';
+
+import { Map, OrderedMap, getIn } from 'immutable';
+
+import reducer, { defaultState, appActions } from './AppReducer';
+
+chai.use(chaiImmutable);
+
+const mockActions = {
+    LOCATION_CHANGE: {
+        type: '@@router/LOCATION_CHANGE',
+        payload: {
+            pathname: 'testPath',
+        },
+    },
+    [appActions.STEEM_API_ERROR]: {
+        type: appActions.STEEM_API_ERROR,
+    },
+    [appActions.FETCH_DATA_BEGIN]: {
+        type: appActions.FETCH_DATA_BEGIN,
+    },
+    [appActions.FETCH_DATA_END]: {
+        type: appActions.FETCH_DATA_END,
+    },
+    [appActions.ADD_NOTIFICATION]: {
+        type: appActions.ADD_NOTIFICATION,
+        payload: {
+            key: 'testKey',
+        },
+    },
+    [appActions.REMOVE_NOTIFICATION]: {
+        type: appActions.REMOVE_NOTIFICATION,
+        payload: {
+            key: 'testKey',
+        },
+    },
+    [appActions.UPDATE_NOTIFICOUNTERS]: {
+        type: appActions.UPDATE_NOTIFICOUNTERS,
+        payload: {
+            follow: 1,
+            total: 2,
+        },
+    },
+    [appActions.SET_USER_PREFERENCES]: {
+        type: appActions.SET_USER_PREFERENCES,
+        payload: {
+            cat: 'mymy',
+            dog: 'polly',
+        },
+    },
+    [appActions.TOGGLE_NIGHTMODE]: {
+        type: appActions.TOGGLE_NIGHTMODE,
+    },
+    [appActions.TOGGLE_BLOGMODE]: {
+        type: appActions.TOGGLE_BLOGMODE,
+    },
+};
+
+const key = mockActions[appActions.ADD_NOTIFICATION].payload.key;
+const mockNotification = OrderedMap({
+    [key]: {
+        action: 'missing translation: en.g.dismiss',
+        dismissAfter: 10000,
+        key,
+    },
+});
+
+describe('App reducer', () => {
+    it('should provide a nice initial state', () => {
+        const initial = reducer();
+        expect(initial).to.equal(defaultState);
+    });
+    it('should return correct state for a LOCATION_CHANGE action', () => {
+        const initial = reducer();
+        const actual = reducer(initial, mockActions['LOCATION_CHANGE']);
```

Comment:
why not use the action creator as the reducer input here?

Reply:
if we use the action creator function as input for the reducer test, we get coverage for both functions